src/sources/util/tcp.rs

```diff
@@ -59,6 +58,27 @@ async fn make_listener(
     }
 }
 
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum TcpSourceAck {
+    Ack,
+    Error,
+    Reject,
+}
+
+pub trait TcpSourceAcker {
+    fn build_ack(self, ack: TcpSourceAck) -> Bytes;
```

Comment:
Wonder if it would just be a little simpler to make this return `Option<Bytes>` to demonstrate that not all acknowledgement states end up generating a response to the caller?

Not that `Bytes::new()` is expensive at all, just... seems like it could be slightly cleaner.

Reply:
That makes sense, and lets us skip a null write to the socket too. Done in 9d9fe89fcf6b2623c7bca251145d9bcf3afa7b87